src/app/customer/Customer.tsx

```diff
@@ -49,52 +51,90 @@ export interface WithCheckoutCustomerProps {
     createAccountError?: Error;
     signInError?: Error;
     clearError(error: Error): Promise<CheckoutSelectors>;
-    continueAsGuest(credentials: GuestCredentials): Promise<CheckoutSelectors>;
+    continueAsGuest(credentials: GuestCredentials, options: CustomerRequestOptions): Promise<CheckoutSelectors>;
     deinitializeCustomer(options: CustomerRequestOptions): Promise<CheckoutSelectors>;
     initializeCustomer(options: CustomerInitializeOptions): Promise<CheckoutSelectors>;
+    loadPaymentMethods(): Promise<CheckoutSelectors>;
     sendLoginEmail(params: { email: string }): Promise<CheckoutSelectors>;
-    signIn(credentials: CustomerCredentials): Promise<CheckoutSelectors>;
-    createAccount(values: CustomerAccountRequestBody): Promise<CheckoutSelectors>;
+    signIn(credentials: CustomerCredentials, options: CustomerRequestOptions): Promise<CheckoutSelectors>;
+    createAccount(values: CustomerAccountRequestBody, options: CustomerRequestOptions): Promise<CheckoutSelectors>;
 }
 
 export interface CustomerState {
+    isReady: boolean;
     isEmailLoginFormOpen: boolean;
     hasRequestedLoginEmail: boolean;
 }
 
 class Customer extends Component<CustomerProps & WithCheckoutCustomerProps, CustomerState> {
     state: CustomerState = {
+        isReady: false,
         isEmailLoginFormOpen: false,
         hasRequestedLoginEmail: false,
     };
 
     private draftEmail?: string;
+    private providerIdWithCustomContinueFlow?: string;
 
-    componentDidMount(): void {
+    async componentDidMount(): Promise<void> {
         const {
-            onReady = noop,
             email,
+            loadPaymentMethods,
+            initializeCustomer,
+            onReady = noop,
+            onUnhandledError = noop,
         } = this.props;
 
         this.draftEmail = email;
 
+        try {
+            const { data } = await loadPaymentMethods();
+
+            this.providerIdWithCustomContinueFlow = this.getCustomCustomerFlowProviderId(data.getPaymentMethods());
+
+            if (this.providerIdWithCustomContinueFlow) {
```

Comment:
`initializeCustomer` should be no-op when called with a method that doesn't require initialisation. So you don't need to do this check.